clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocol.java

```diff
@@ -64,19 +64,19 @@
 
     public static final short CONSUMER_PROTOCOL_V0 = 0;
     public static final Schema CONSUMER_PROTOCOL_HEADER_SCHEMA = new Schema(
-            new Field(VERSION_KEY_NAME, Type.INT16));
+            new Field(VERSION_KEY_NAME, Type.INT16, "Version number of the consumer protocol."));
     private static final Struct CONSUMER_PROTOCOL_HEADER_V0 = new Struct(CONSUMER_PROTOCOL_HEADER_SCHEMA)
             .set(VERSION_KEY_NAME, CONSUMER_PROTOCOL_V0);
 
     public static final Schema SUBSCRIPTION_V0 = new Schema(
-            new Field(TOPICS_KEY_NAME, new ArrayOf(Type.STRING)),
-            new Field(USER_DATA_KEY_NAME, Type.NULLABLE_BYTES));
+            new Field(TOPICS_KEY_NAME, new ArrayOf(Type.STRING), "The topics the consumer subscribes to."),
```

Comment:
Drop "The .. the .. " -> "Topics consumer is subscribing to" or "Consumer's topic subscription"

Same for rest..

Reply:
Thanks for your suggestions. I plan to add description to most of the fields and ping you once I finished. Unfortunately I have to redo all [this](https://github.com/apache/kafka/compare/trunk...andrasbeni:before-rework) because of an other PR merged in the meantime.